changelog/15946.txt

```diff
@@ -0,0 +1,3 @@
+```release-note:bug
+core/seal: write a temp file and encrypt it, then rename it to the target path file. This can avoid the orignal file crash if Vault restart.
```

Comment:
This changelog entry should indicate what changed from a user's perspective. For example, in this case, maybe something like:
```
Fix possible keyring truncation when using the file backend.
```

Reply:
Ok. I am new in github,  If I still needs to update the file in my repository (Vault fork project) ?  How can I squash this pull request(There are three commits so far). It seems no branch name and I have no permission.

Reply:
Anyway I update the file in the fouth commit.

Reply:
Yeah, no need to worry about squashing. That will happen automatically when the PR is merged.